backend-api/src/functions/services/session/tests/sessionService.test.ts

```diff
@@ -0,0 +1,304 @@
+import {
+  ConditionalCheckFailedException,
+  DynamoDBClient,
+  DynamoDBClientResolvedConfig,
+  PutItemCommand,
+  QueryCommand,
+  ServiceInputTypes,
+  ServiceOutputTypes,
+} from "@aws-sdk/client-dynamodb";
+import { AwsStub, mockClient } from "aws-sdk-client-mock";
+import { ISessionService, SessionService } from "../sessionService";
+
+describe("Session Service", () => {
+  let sessionService: ISessionService;
+  let dynamoDbMockClient: AwsStub<
+    ServiceInputTypes,
+    ServiceOutputTypes,
+    DynamoDBClientResolvedConfig
+  >;
+
+  beforeEach(() => {
+    sessionService = new SessionService("mockTableName");
+    dynamoDbMockClient = mockClient(DynamoDBClient);
+  });
+
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date("2024-03-10"));
+  });
+
+  describe("Get active session ID", () => {
```

Comment:
we should explicitly test the scenario where the query returns two sessions I think

Reply:
It won't return two sessions because the query includes `Limit: 1`

Reply:
Thank you, yes. 

I supposed I was thinking about the fact we do items[0]. if we changed it to Limit:2 we wouldn't have a test that fails.

But I'm happy with what we've got

Reply:
And I assume this is why you want jest mocks!